paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/clone/CopyFileOperator.java

```diff
@@ -139,6 +141,37 @@ private Path pathOfTable(Table table) {
         return new Path(table.options().get(CoreOptions.PATH.key()));
     }
 
+    private void copyManifestFile(Path sourcePath, Path targetPath, CloneFileInfo cloneFileInfo)
+            throws IOException, TableNotExistException {
+        Identifier sourceIdentifier = Identifier.fromString(cloneFileInfo.getSourceIdentifier());
+        FileStoreTable sourceTable = (FileStoreTable) sourceCatalog.getTable(sourceIdentifier);
+        FileStore<?> store = sourceTable.store();
+        ManifestFile manifestFile = store.manifestFileFactory().create();
+
+        List<ManifestEntry> manifestEntries =
+                manifestFile.readWithIOException(sourcePath.getName());
+        List<ManifestEntry> targetManifestEntries = new ArrayList<>(manifestEntries.size());
+
+        // clone job will clone the source path to target warehouse path, so the target external
+        // path is null
+        String newExternalPath = null;
+        for (ManifestEntry manifestEntry : manifestEntries) {
+            ManifestEntry newManifestEntry =
+                    new ManifestEntry(
+                            manifestEntry.kind(),
+                            manifestEntry.partition(),
+                            manifestEntry.bucket(),
+                            manifestEntry.totalBuckets(),
+                            manifestEntry.file().copy(newExternalPath));
```

Comment:
Value 'newExternalPath' is always 'null'

Reply:
Yes, the same as above

Reply:
ditto